src/Entity/OgRoleListBuilder.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\og\Entity;
+
+use Drupal\Core\Url;
+use Drupal\Core\Config\Entity\DraggableListBuilder;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\og\Og;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Defines a class to build a listing of og_role entities.
+ *
+ * @see \Drupal\og\Entity\OgRole
+ */
+class OgRoleListBuilder extends DraggableListBuilder {
+
+  /**
+   * The group entity type.
+   *
+   * @var string
+   */
+  protected $groupType;
+
+  /**
+   * The group entity bundle id.
+   *
+   * @var string
+   */
+  protected $groupBundle;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function createInstance(ContainerInterface $container, EntityTypeInterface $entity_type) {
+    return new static(
+      $entity_type,
+      $container->get('entity_type.manager')->getStorage($entity_type->id()),
+      $container->get('current_route_match')
+    );
+  }
+
+  /**
+   * Constructs a new OgRoleListBuilder object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeInterface $entity_type
+   *   The entity type definition.
+   * @param \Drupal\Core\Entity\EntityStorageInterface $storage
+   *   The entity storage class.
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The route matcher.
+   */
+  public function __construct(EntityTypeInterface $entity_type, EntityStorageInterface $storage, RouteMatchInterface $route_match) {
+    parent::__construct($entity_type, $storage);
+
+    // When on the default og_role list route,
+    // we should have a group entity type and bundle.
+    if ($route_match->getRouteName() == 'entity.og_role.collection') {
+      $parameters = $route_match->getParameters();
+
+      // Check if the route had a group type parameter.
+      if ($parameters->has('entity_type_id')) {
+        $this->groupType = $parameters->get('entity_type_id');
+      }
+      if ($parameters->has('bundle_id')) {
+        $this->groupBundle = $parameters->get('bundle_id');
+      }
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEntityIds() {
+    $query = $this->getStorage()->getQuery()
+      ->condition('group_type', $this->groupType, '=')
+      ->condition('group_bundle', $this->groupBundle, '=')
+      ->sort($this->entityType->getKey('weight'));
```

Comment:
Add access check to entity query for Drupal 10 compatibility.
```suggestion
      ->sort($this->entityType->getKey('weight'))
      ->accessCheck();
```